modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -201,231 +208,129 @@
 
-    public static BValue[] setStringPayload(Context context,
-            AbstractNativeFunction abstractNativeFunction, boolean isRequest) {
-        BStruct httpMessageStruct = (BStruct) abstractNativeFunction.getRefArgument(context, 0);
         HTTPCarbonMessage httpCarbonMessage = HttpUtil
                 .getCarbonMsg(httpMessageStruct, HttpUtil.createHttpCarbonMessage(isRequest));
-
         httpCarbonMessage.waitAndReleaseAllEntities();
 
-        String payload = abstractNativeFunction.getStringArgument(context, 0);
         OutputStream messageOutputStream = new HttpMessageDataStreamer(httpCarbonMessage).getOutputStream();
-        StringDataSource stringDataSource = new StringDataSource(payload, messageOutputStream);
-        addMessageDataSource(httpMessageStruct, stringDataSource);
-        addMessageOutputStream(httpMessageStruct, messageOutputStream);
-
-        HttpUtil.setHeaderToStruct(context, httpMessageStruct, Constants.CONTENT_TYPE, Constants.TEXT_PLAIN);
-        if (log.isDebugEnabled()) {
-            log.debug("Setting new payload: " + payload);
+        BStruct entity = (BStruct) abstractNativeFunction.getRefArgument(context, ENTITY_INDEX);
+        String baseType = getContentType(entity);
+        boolean isBodyAvailable;
+
+        if (baseType != null) {
+            switch (baseType) {
+                case org.ballerinalang.mime.util.Constants.TEXT_PLAIN:
+                    isBodyAvailable = MimeUtil.isTextBodyPresent(entity);
+                    break;
+                case org.ballerinalang.mime.util.Constants.APPLICATION_JSON:
+                    isBodyAvailable = MimeUtil.isJsonBodyPresent(entity);
+                    break;
+                case org.ballerinalang.mime.util.Constants.APPLICATION_XML:
+                    isBodyAvailable = MimeUtil.isXmlBodyPresent(entity);
+                    break;
+                default:
+                    isBodyAvailable = MimeUtil.isBinaryBodyPresent(entity);
+                    break;
+            }
+        } else {
+            baseType = org.ballerinalang.mime.util.Constants.OCTET_STREAM;
+            isBodyAvailable = MimeUtil.isBinaryBodyPresent(entity);
+        }
+        HttpUtil.setHeaderToStruct(context, entity, CONTENT_TYPE, baseType);
+        httpMessageStruct.addNativeData(MESSAGE_ENTITY, entity);
+        if (isBodyAvailable) {
+            httpMessageStruct.addNativeData(IS_ENTITY_BODY_PRESENT, true);
         }
+        addMessageOutputStream(httpMessageStruct, messageOutputStream);
         return AbstractNativeFunction.VOID_RETURN;
     }
 
-    public static BValue[] setXMLPayload(Context context,
-            AbstractNativeFunction abstractNativeFunction, boolean isRequest) {
-        BStruct httpMessageStruct = (BStruct) abstractNativeFunction.getRefArgument(context, 0);
+    /**
+     * Get the entity from request or response.
+     *
+     * @param context                Ballerina context
```

Comment:
Shall we remove the additional spaces ? Because we have't followed similar convention in other places. We have to stick to one convention.

Reply:
This is in accordance with the WSO2 coding style for java docs. It automatically aligns the parameters. Maybe the other places don't follow the correct coding style.